corehq/form_processor/migrations/0084_migrate_balance_1_add_column.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from corehq.sql_db.operations import RawSQLMigration
+
+
+migrator = RawSQLMigration(('corehq', 'form_processor', 'migrations', 'sql_templates'))
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('form_processor', '0083_migrate_delta_alter_field'),
+    ]
+
+    operations = [
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrator.get_migration('migrate_balance_1_add_column.sql'),
```

Comment:
I was under the impression that Django runs each migration operation in a separate transaction. If that's the case, all migrations added in this PR could be merged into a single migration with multiple operations. Something like:
```py
    operations = [
        migrations.SeparateDatabaseAndState(
            database_operations=[
                migrator.get_migration('migrate_balance_1_add_column.sql'),
                migrator.get_migration('migrate_balance_2_create_trigger.sql'),
                migrator.get_migration('migrate_balance_3_backfill_notnull.sql'),
                migrator.get_migration('migrate_balance_4_switch_columns.sql'),
            ]
        ),
        migrations.SeparateDatabaseAndState(
            state_operations=[
                migrations.AlterField(
                    model_name='ledgertransaction',
                    name='updated_balance',
                    field=models.BigIntegerField(default=0),
                ),
            ]
        ),
    ]
```
Would that work?

Reply:
Hmm, I think I am wrong. The documentation isn't very clear on this, but it looks like each migration (not each operation) is run in a separate transaction:

> On databases that support DDL transactions (SQLite and PostgreSQL), migrations will run inside a transaction by default. ([source](https://docs.djangoproject.com/en/2.1/howto/writing-migrations/#non-atomic-migrations))

so I think the way you have it split up is the way it has to be.

Alternatively, if you want to group them into a single migration like I suggested above you could use `atomic = False` on the mitration:
```py
class Migration(migrations.Migration):
    atomic = False
```
and do explicit transaction management with `BEGIN`/`COMMIT` statements inside each of the SQL files.

Reply:
Ah! I'll leave this as it is, but that is interesting.